### ISSUE TITLE
npm run dev errors when running in vagrant

### ISSUE THREAD

**Issue by medelasri** [vuejs-templates/webpack#371]
I need to start the dev server using `npm run dev` but I get this errors:

```
vagrant@med:~/modal.dev$ npm run dev

> modaljs@1.0.0 dev /home/vagrant/modal.dev
> node build/dev-server.js
/home/vagrant/modal.dev/node_modules/express/lib/application.js:370
  return

SyntaxError: Invalid or unexpected token
    at Object.exports.runInThisContext (vm.js:76:16)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/vagrant/modal.dev/node_modules/express/lib/expr    ess.js:17:13)
    at Module._compile (module.js:570:32)

npm ERR! Linux 3.2.0-23-generic-pae
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "dev"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! modaljs@1.0.0 dev: `node build/dev-server.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the modaljs@1.0.0 dev script 'node build/dev-server.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the modaljs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build/dev-server.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs modaljs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls modaljs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/vagrant/modal.dev/npm-debug.log
```

The first one is fixed by updating the express package to 4.14.0 but I get another error:

```
vagrant@med:~/modal.dev$ npm run dev

> modaljs@1.0.0 dev /home/vagrant/modal.dev
> node build/dev-server.js

/home/vagrant/modal.dev/node_modules/content-disposition/index.js:13
module.exports = contentDisposition
                 ^

ReferenceError: contentDisposition is not defined
    at Object.<anonymous> (/home/vagrant/modal.dev/node_modules/content-disposit    ion/index.js:13:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/vagrant/modal.dev/node_modules/express/lib/util    s.js:15:26)
    at Module._compile (module.js:570:32)

npm ERR! Linux 3.2.0-23-generic-pae
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "dev"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! modaljs@1.0.0 dev: `node build/dev-server.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the modaljs@1.0.0 dev script 'node build/dev-server.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the modaljs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build/dev-server.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs modaljs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls modaljs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/vagrant/modal.dev/npm-debug.log
```

**Comment by LinusBorg** [comment]
Do you have the same problems when you try using it outside of vagrant?

Seems it's related to vagrant - a guide to reproduce this would be helpful.

**Comment by necevil** [comment]
I had a similar issue in Docker that was solved by removing my node_modules directory with and doing fresh `npm install`

Not sure if you have tried that yet but its pretty painless and might clear it up without messing with your local environment.

**Comment by LinusBorg** [comment]
closing for inactivity.

**Comment by aankittcoolest** [comment]
I am also facing the same issue!
Here is the detailed log. I think this can be of some use.

<pre>
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/bin/npm', 'run', 'dev' ]
2 info using npm@3.10.10
3 info using node@v6.9.4
4 verbose run-script [ 'predev', 'dev', 'postdev' ]
5 info lifecycle client_labs@1.0.0~predev: client_labs@1.0.0
6 silly lifecycle client_labs@1.0.0~predev: no script for predev, continuing
7 info lifecycle client_labs@1.0.0~dev: client_labs@1.0.0
8 verbose lifecycle client_labs@1.0.0~dev: unsafe-perm in lifecycle true
9 verbose lifecycle client_labs@1.0.0~dev: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/home/vagrant/client_labs/node_modules/.bin:/usr/bin:/sbin:/bin:/usr/sbin:/usr/bin
10 verbose lifecycle client_labs@1.0.0~dev: CWD: /home/vagrant/client_labs
11 silly lifecycle client_labs@1.0.0~dev: Args: [ '-c', 'node build/dev-server.js' ]
12 silly lifecycle client_labs@1.0.0~dev: Returned: code: 1  signal: null
13 info lifecycle client_labs@1.0.0~dev: Failed to exec dev script
14 verbose stack Error: client_labs@1.0.0 dev: `node build/dev-server.js`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:255:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:877:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid client_labs@1.0.0
16 verbose cwd /home/vagrant/client_labs
17 error Linux 3.10.0-327.36.2.el7.x86_64
18 error argv "/usr/bin/node" "/bin/npm" "run" "dev"
19 error node v6.9.4
20 error npm  v3.10.10
21 error code ELIFECYCLE
22 error client_labs@1.0.0 dev: `node build/dev-server.js`
22 error Exit status 1
23 error Failed at the client_labs@1.0.0 dev script 'node build/dev-server.js'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the client_labs package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     node build/dev-server.js
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs client_labs
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls client_labs
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

</pre>

**Comment by necevil** [comment]
@aankittcoolest Just a heads up, when I ran into a similar issue with Docker it ended up being a permissions problem inside of the container.  I had to add a chown line to my docker compose file to make sure the user running the dev server inside of the container had proper permissions.

I see that you are working with Vagrant but figured I would chime in and let you know what ended up straightening out my problem (though it could easily be different for you with Vagrant).
